File: ProjetAngular/src/app/pagesweb/model/partie.ts
import { Personnage } from './personnage';
import { Inventaire } from './inventaire';
import { Events } from './events';
import { Environnement } from './environnement';
import { Compte } from './compte';
export class Partie {
  public get personnage(): Personnage | undefined {
    return this._personnage;
  }
  public set personnage(value: Personnage | undefined) {
    this._personnage = value;
  }
  public get date(): Date | undefined {
    return this._date;
  }
  public set date(value: Date | undefined) {
    this._date = value;
  }
  public get inventaire(): Inventaire | undefined {
    return this._inventaire;
  }
  public set inventaire(value: Inventaire | undefined) {
    this._inventaire = value;
  }
  public get eventRunning(): Events | undefined {
    return this._eventRunning;
  }
  public set eventRunning(value: Events | undefined) {
    this._eventRunning = value;
  }
  public get environment(): Environnement | undefined {
    return this._environment;
  }
  public set environment(value: Environnement | undefined) {
    this._environment = value;
  }
  public get compte(): Compte | undefined {
    return this._compte;
  }
  public set compte(value: Compte | undefined) {
    this._compte = value;
  }
  public get id(): number | undefined {
    return this._id;
  }
  public set id(value: number | undefined) {
    this._id = value;
  }
  constructor(
    private _id?: number,
    private _compte?: Compte,
    private _environment?: Environnement,
    private _eventRunning?: Events,
    private _inventaire?: Inventaire,
    private _personnage?: Personnage,
    private _date?: Date
  ) {}
}

File: projetjeuSpringBoot/projetjeu/src/main/java/fr/projetjeu/model/Partie.java
package fr.projetjeu.model;

import java.sql.Timestamp;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name="partie")
public class Partie {

	@JsonView(JsonViews.Common.class)
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "par_id")
	private Integer id;
	
	@JsonView(JsonViews.Partie.class)
	@Column(name="par_date", nullable=false)
	private Timestamp date;
	
	@JsonView(JsonViews.Partie.class)
	@OneToOne
	@JoinColumn(name="par_personnage_id", nullable = false)
	private Personnage personnage;
	
	@JsonView(JsonViews.Partie.class)
	@ManyToOne
	@JoinColumn(name="par_event_id", nullable = false)
	private Events eventRunning;
	
	@JsonView(JsonViews.Partie.class)
	@ManyToOne
	@JoinColumn(name="par_environnement_id", nullable=false)
	private Environnement environnement;
	
	@JsonView(JsonViews.Partie.class)
	@OneToOne
	@JoinColumn(name="par_inventaire_id", nullable=false)
	private Inventaire inventaire;
	
	@ManyToOne
	@JoinColumn(name="par_compte_id",nullable=false)
	private Compte compte;


	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public Personnage getPersonnage() {
		return personnage;
	}

	public void setPersonnage(Personnage personnage) {
		this.personnage = personnage;
	}

	public Events getEventRunning() {
		return eventRunning;
	}

	public void setEventRunning(Events eventRunning) {
		this.eventRunning = eventRunning;
	}

	public Timestamp getDate() {
		return date;
	}

	public void setDate(Timestamp date) {
		this.date = date;
	}

	public Environnement getEnvironnement() {
		return environnement;
	}

	public void setEnvironnement(Environnement environnement) {
		this.environnement = environnement;
	}

	public Inventaire getInventaire() {
		return inventaire;
	}

	public void setInventaire(Inventaire inventaire) {
		this.inventaire = inventaire;
	}

	public Compte getCompte() {
		return compte;
	}

	public void setCompte(Compte compte) {
		this.compte = compte;
	}
	
	
	
}
